include/seqan3/range/container/dynamic_bitset.hpp

```diff
@@ -0,0 +1,1479 @@
+     * \throws std::invalid_argument if any character is not '0' or '1'.
+     *
+     * The `char` literal is expected to be null-terminated (asserted in debug-mode). If it is not, the last character
+     * will be lost when copying to the instance of `dynamic_bitset`.
+     *
+     * ### Complexity
+     *
+     * Linear in the size of _lit.
+     *
+     * ### Exceptions
+     *
+     * Throws std::invalid_argument if any character is not '0' or '1'.
+     */
+    template <size_t N>
+    constexpr void assign(char const (&_lit)[N])
+    {
+        static_assert(N <= capacity_ + 1, "Length of string literal exceeds capacity of dynamic_bitset.");
+        assert(_lit[N - 1] == '\0');
+        uint64_t value{};
+        for (size_t i = 0; i != N - 1; ++i)
+        {
+            if (_lit[i] == '0')
+            {
+                value <<= 1;
+            }
+            else if (_lit[i] == '1')
+            {
+                if (i != 0)
+                {
+                    value <<= 1;
+                }
+                value |= 1u;
+            }
+            else
+            {
+                throw std::invalid_argument{"The string to construct a dynamic_bitset from May only contain 0 and 1."};
+            }
+        }
+
+        (*this) = value;
+        resize(N - 1);
+    }
+
+    /*!\brief Assign from `std::initializer_list`.
+     * \param[in] ilist A `std::initializer_list` of value_type.
+     *
+     * ### Complexity
+     *
+     * Linear in the size of `ilist`.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    constexpr void assign(std::initializer_list<value_type> ilist) noexcept
+    {
+        assign(std::ranges::begin(ilist), std::ranges::end(ilist));
+    }
+
+    /*!\brief Assign with `count` times `value`.
+     * \param[in] count Number of elements.
+     * \param[in] value The initial value to be assigned.
+     *
+     * ### Complexity
+     *
+     * In \f$O(count)\f$.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    constexpr void assign(size_type const count, value_type const value) noexcept
+    {
+        clear();
+        auto tmp = view::repeat_n(value, count);
+        assign(std::ranges::begin(tmp), std::ranges::end(tmp));
+    }
+
+    /*!\brief Assign from a different range.
+     * \tparam other_range_t The type of range to be inserted; must satisfy std::ranges::InputRange and `value_type`
+     *                       must be constructible from reference_t<other_range_t>.
+     * \param[in]      range The sequences to construct/assign from.
+     *
+     * ### Complexity
+     *
+     * Linear in the size of `range`.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    template <std::ranges::InputRange other_range_t>
+    //!\cond
+        requires std::Constructible<value_type, /*ranges::range_reference_t*/reference_t<other_range_t>>
+    //!\endcond
+    constexpr void assign(other_range_t && range) noexcept
```

Comment:
@h-2 You said that containers will not get a range constructor in the standard, so we also don't need it, right?

Reply:
If so, I'd rather do another PR where this is changed in all the containers we provide.

Reply:
kay

Reply:
I also added a card, so we don't forget.